token/core/zkatdlog/crypto/common/array.go

```diff
@@ -22,18 +23,16 @@ type G1Array []*math.G1
 
 // Bytes serialize an array of G1 elements
 func (a *G1Array) Bytes() ([]byte, error) {
-
-	var raw []byte
-	for _, e := range []*math.G1(*a) {
+	raw := make([][]byte, 2*len([]*math.G1(*a)))
+	for i, e := range []*math.G1(*a) {
 		if e == nil {
 			return nil, errors.Errorf("failed to marshal array of G1")
 		}
 		st := hex.EncodeToString(e.Bytes())
-		raw = append(raw, []byte(Separator)...)
-		raw = append(raw, []byte(st)...)
-
+		raw[2*i] = []byte(st)
+		raw[2*i+1] = []byte(Separator)
 	}
-	return raw, nil
+	return json.Marshal(raw)
```

Comment:
better to avoid json here given that it is slow.
What you could do is to obtain the 3 slices you need and then aggregate them by allocating a new slice of size the sum of the 3 slices. Then you can use append.

Reply:
Hi @adecaro, I will use `bytes.Join(raw, separator)` instead. It seems to be fit for purpose.